GCEWindowsAgent/serial_listener.go

```diff
@@ -0,0 +1,196 @@
+//  Copyright 2019 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os/exec"
+	"path"
+	"syscall"
+	"time"
+
+	"github.com/GoogleCloudPlatform/guest-logging-go/logger"
+)
+
+type serialMessage interface{}
```

Comment:
empty interfaces are rarely useful outside of reflection. if you really think you need an interface (i hate them and avoid them at all costs), you should give it definitional methods